### DISCUSSION TITLE
Need all files associated with a VM in one directory

### DISCUSSION THREAD

**Discussion by blakemcbride** [virt-manager/virt-manager#656] [General]
I use VM's on a pretty regular basis. My preference is to use KVM/QEMU/virt-manager. However, I have found virt-manager unusable for me for the following three reasons.

First, With all of the other VM systems (e.g., VirtualBox), all of the files necessary for a single VM get put in its own directory. This means I can easily know where all of the files are, back them up, and move individual VMs from one host to another. Virt-manager seems to be a tack-on to KVM et al. For each VM, files are in several places.  It makes things complicated and error prone to manage.

Second, multiple VMs share common config files, meaning I can't split one VM from another.

The third problem I have with virt-manager is the whole file volume thing. I suppose it makes sense in a high-volume production environment, but it makes no sense at all in a single-developer system. It just adds needless complexity and hassle.

All of this makes virt-manager unusable for me.

I definitely appreciate all of the work done on KVM/QEMU and virt-manager. I just think it needs some tweaking before it is usable for me. Perhaps others share some of these challenges.

**Comment by phrdina** [comment]
Hi, sorry for taking that long to reply, the simple answer is this cannot be changed in virt-manager because it is GUI using libvirt under the hood to manage QEMU/KVM VMs.

**Comment by blakemcbride** [comment]
Your GUI can be made to package up all of those disparate files for backup, restore, or move purposes. You can also add a delete function that deletes all of the entries for a single VM from the host. Adding this code would be relatively easy and would immensely improve virt-manager's value.
